README.md

```diff
@@ -0,0 +1,71 @@
+# Minimal ROS 2 real-time cookbook recipes
+
+Collection of ROS 2 minimal examples showing how to leverage different POSIX and ROS 2 features to
+create real-time applications.
+
+
+* [minimal_memory_lock](minimal_memory_lock/README.md): shows how to lock the process memory and
+ pre-allocate dynamic memory to avoid memory page faults
+
+
+TODO: Add other packages after review (currently in rolling-experimental branch): 
```

Comment:
As a note, I'm happy to help reviewing all of these as well, as I'm working on similar stuff at the moment.

Reply:
>I would consider just calling mlockall all the time and not be able to turn it off. 

This is how it was implemented at the begging. Then, I started debugging I wanted to compare with and without memory lock. In fact, I think is very helpful for the users to compare both cases. 

I see the following options:
1. Remove all the options, keep it minimal
2. Keep all the options. Add comments explaining the memory has to be locked for real-time behavior. 
3. Create a separate executable without locking the memory

@shuhaowu @LanderU What do you think?

Reply:
>As a note, I'm happy to help review all of these as well, as I'm working on similar stuff at the moment.

Let me know which ones you want to review and I will create PRs for those. I can create PRs for scheduling and affinity. Is this okay?

Reply:
Sure. I'm happy to review all of them in any order you wish!

Reply:
@carlossvg,

+1 to

> Keep all the options. Add comments explaining the memory has to be locked for real-time behavior.

Reply:
Actually, I'm going to show some prints explaining what to expect when we don't lock the memory.

Reply:
Added prints